builtin/game/item.lua

```diff
@@ -503,50 +524,63 @@ function core.handle_node_drops(pos, drops, digger)
 				core.add_item(p, left)
 			end
 		end
+	else
+		for _, dropped_item in ipairs(drops) do
+			local p = {
+				x = pos.x + math.random()/2-0.25,
+				y = pos.y + math.random()/2-0.25,
+				z = pos.z + math.random()/2-0.25,
+			}
+			core.add_item(p, dropped_item)
+		end
```

Comment:
```Lua
-- Add dropped items to object's inventory
local inv = digger and digger:get_inventory()
for _, dropped_item in pairs(drops) do
	local left = inv and inv:add_item("main", dropped_item) or dropped_item
	if not inv or not left:is_empty() then
		-- spawn code
	end
end
```
Covers both cases without code duplication. (untested)

Reply:
But that would require doing the inventory check every loop counter, which is probably ok considering how many times it will run, but feels ugly. I think I know another way to remove the duplication though.

Reply:
Such a check isn't expensive at all, similar to an `if` expression with the `inv` condition.